tests/test_inference.py

```diff
@@ -3781,6 +3811,36 @@ def test_standard_pipeline(self):
                 sample = inf_ts.node(u)
                 assert sample.time == t
 
+    def test_insert_proxy_time_units_mismatch(self):
+        samples = [msprime.Sample(population=0, time=t) for t in (0.0, 0.0, 0.1, 1.5)]
```

Comment:
I don't find this test particularly enlightening - it hinges on me knowing precisely what ``from_tree_sequence`` does as a side effect in terms of the ``time_units`` variable. We should be aiming for tests that someone can read in isolation. There's no point in simulating a bunch of data here as we're never testing the output of that process.